pulsar-client/src/main/java/org/apache/pulsar/client/impl/PartitionedProducerImpl.java

```diff
@@ -76,7 +77,11 @@ private MessageRouter getMessageRouter() {
             messageRouter = customMessageRouter;
             break;
         case RoundRobinPartition:
-            messageRouter = new RoundRobinPartitionMessageRouterImpl(conf.getHashingScheme());
+            messageRouter = new RoundRobinPartitionMessageRouterImpl(
+                conf.getHashingScheme(),
+                ThreadLocalRandom.current().nextInt(topicMetadata.numPartitions()),
+                conf.isBatchingEnabled(),
+                TimeUnit.MICROSECONDS.toMillis(conf.getBatchingMaxPublishDelayMicros()));
```

Comment:
If the `maxPublishDelay is < 1millis`, this would get converted into a 0. Later when we do `currentMs / maxBatchingDelayMs` we would get divide by 0.

Reply:
fixed by doing "Math.max(1, ...)"